### DISCUSSION TITLE
Can't use select machine with multiple in form

### DISCUSSION THREAD

**Discussion by azoriusss** [chakra-ui/zag#1301] [Q&A] [answered]
Hi
I've been using zag for a few days and I've found an issue with multiple select

When using select machine with `multiple` option enabled, everything seems to work fine (on state level at least). But when this select is wrapped inside a form, it seems that selected values are not sent in url  on submit (like if they were not selected in the hidden select)
It works fine when only selecting only one option 

Is there something I'm missing here ? or is this a bug ?

Here's a [codesandbox](https://codesandbox.io/p/sandbox/quizzical-merkle-l7ynh3?file=%2Fsrc%2FSelect.tsx%3A9%2C21&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt5tojy500062v6h1cbp0boo%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt5tojy500022v6hknebrnib%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt5tojy500032v6hu8efomow%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt5tojy500052v6hrd2fveke%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt5tojy500022v6hknebrnib%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt5tojy400012v6huljnep0t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A9%252C%2522startColumn%2522%253A2%252C%2522endLineNumber%2522%253A11%252C%2522endColumn%2522%253A22%257D%255D%257D%252C%257B%2522id%2522%253A%2522clt5ttq6k00022v6he6hsawmu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A9%252C%2522startColumn%2522%253A21%252C%2522endLineNumber%2522%253A9%252C%2522endColumn%2522%253A21%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FSelect.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clt5tojy500022v6hknebrnib%2522%252C%2522activeTabId%2522%253A%2522clt5ttq6k00022v6he6hsawmu%2522%257D%252C%2522clt5tojy500052v6hrd2fveke%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt5tojy500042v6hqgozqfi9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clt5tojy500052v6hrd2fveke%2522%252C%2522activeTabId%2522%253A%2522clt5tojy500042v6hqgozqfi9%2522%257D%252C%2522clt5tojy500032v6hu8efomow%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt5tojy500032v6hu8efomow%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Note : I've also tried to use ArkUI to achieve this but I'm getting the same issue.

Thank you !

**Comment by TylerAPfledderer** [comment]
This is normal behavior from the native `select` element, which the hidden form element as part of the machine.

So if you do something like this (basic example):
```tsx
<form>
   <select multiple>
     <option value="parrot">Parrot</option>
     <option value="macaw">Macaw</option>
     <option value="albatross">Albatross</option>
   </select>
   <button type="submit">Submit</button>
 </form>
```

You will still only see a blank query in the url.

Not sure if this is within the scope of the machine, and rather in the scope of how the form is created and controlled. I would think that it would depend on the user in how they want to format the query with multiple options.

If I recall correctly, these machines are meant to be built on top of native functionality to meet accessibility guidelines and flexibility in design, and not to do anything more with handling data outside of ensuring that machines function.

**Reply by azoriusss** [reply]
Thank you for your answer
The exemple you gave will indeed show a blank query, but that's because the select has no declared name
But anyway, I can see the point you're making with the purpose of machine

I've found a way to handle my use case like this 
`<select value={api.value} onChange={() => {}} {...api.hiddenSelectProps}> `

**Reply by segunadebayo** [reply, marked as answer]
This was indeed a bug. I just pushed a fix for this
